From 7a069ee3ce55834c459a5f7b4dfa21fb3b292557 Mon Sep 17 00:00:00 2001
From: Reedy <reedy@wikimedia.org>
Date: Wed, 02 Dec 2020 01:27:22 +0000
Subject: [PATCH] Stop setting CURLMOPT_PIPELINING on curl >= 7.62.0

Bug: T264735
Change-Id: Ic878367bdfe16bfc2e1cff9fbc6b497ec1fc366b
---

diff --git a/includes/libs/http/MultiHttpClient.php b/includes/libs/http/MultiHttpClient.php
index 750b1dd..35aa579 100644
--- a/includes/libs/http/MultiHttpClient.php
+++ b/includes/libs/http/MultiHttpClient.php
@@ -439,8 +439,10 @@
 			$this->cmh = $cmh;
 		}
 
+		$curlVersion = curl_version()['version'];
+
 		// CURLMOPT_MAX_HOST_CONNECTIONS is available since PHP 7.0.7 and cURL 7.30.0
-		if ( version_compare( curl_version()['version'], '7.30.0', '>=' ) ) {
+		if ( version_compare( $curlVersion, '7.30.0', '>=' ) ) {
 			// Limit the number of in-flight requests for any given host
 			$maxHostConns = $opts['maxConnsPerHost'] ?? $this->maxConnsPerHost;
 			curl_multi_setopt( $this->cmh, CURLMOPT_MAX_HOST_CONNECTIONS, (int)$maxHostConns );
@@ -448,7 +450,15 @@
 
 		// Configure when to multiplex multiple requests onto single TCP handles
 		$pipelining = $opts['usePipelining'] ?? $this->usePipelining;
-		curl_multi_setopt( $this->cmh, CURLMOPT_PIPELINING, $pipelining ? 3 : 0 );
+
+		if ( !$pipelining ) {
+			curl_multi_setopt( $this->cmh, CURLMOPT_PIPELINING, CURLPIPE_NOTHING );
+		} elseif ( version_compare( $curlVersion, '7.62.0', '<' ) ) {
+			// As of cURL 7.62.0, CURLPIPE_HTTP1 (value 1) is no longer supported,
+			// so only set it on versions < 7.62.0.
+			// On versions >= 7.62.0, CURLPIPE_MULTIPLEX is set by default
+			curl_multi_setopt( $this->cmh, CURLMOPT_PIPELINING, CURLPIPE_HTTP1 | CURLPIPE_MULTIPLEX );
+		}
 
 		return $this->cmh;
 	}
