From b7909f146a8d99066dbc57758528352b0ba2f4a0 Mon Sep 17 00:00:00 2001
From: Florian <florian.schmidt.welzow@t-online.de>
Date: Tue, 22 Sep 2020 17:05:21 +0200
Subject: [PATCH] [PATCH] Add custom code for LangLinkHandler because of
 T172076

---
 client/includes/Hooks/LangLinkHandler.php | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/client/includes/Hooks/LangLinkHandler.php b/client/includes/Hooks/LangLinkHandler.php
index e39d772de5..32cb6c8a91 100644
--- a/client/includes/Hooks/LangLinkHandler.php
+++ b/client/includes/Hooks/LangLinkHandler.php
@@ -335,15 +335,8 @@ class LangLinkHandler {
 	 * @return string
 	 */
 	public function getInterwikiCodeFromSite( Site $site ) {
-		// FIXME: We should use $site->getInterwikiIds, but the interwiki ids in
-		// the sites table are wrong currently, see T137537.
-		$id = $site->getGlobalId() ?? '';
-		$id = preg_replace( '/(wiki\w*|wiktionary)$/', '', $id );
-		$id = strtr( $id, [ '_' => '-' ] );
-		if ( !$id ) {
-			$id = $site->getLanguageCode() ?? '';
-		}
-		return $id;
+		// droidwiki.org customisation, see T172076
+		return $site->getLanguageCode();
 	}
 
 }
-- 
2.26.2.windows.1

